### ISSUE TITLE
Recent documents via right arrow not working on High Sierra

### ISSUE THREAD

**Issue by rosano** [quicksilver/Quicksilver#2431]
I found this thread https://groups.google.com/forum/#!msg/blacktree-quicksilver/ccEoDmZMU3k/nhLiWB50BAAJ after experiencing the same issue and I tried Rob's first suggestion to replicate the bug. Here's a short video https://www.dropbox.com/s/ugbz1cr94k7ra2a/1%20replicate.mp4?dl=0

- I made a new user account
- opened 4 PDFs
- confirmed they were in the Open Recent menu item of Preview
- tried to access them from QS via right arrow on Preview (in the video there is no visual feedback when the right arrow doesn't work, but I had pressed it several times)

When it worked in the past I think there used to be a disclosure arrow on the selected application (if it had recent documents), but now there isn't one.

I am using QS 1.5.8 on High Sierra 10.13.3

**Comment by skurfer** [comment]
Try this custom build of the app. It’s identical to the current release, but I’ve added some logging in the recent documents code. Look in the console for lines containing “GET_RECENTS”.

https://cl.ly/2A1x0j190N42

**Comment by mdaeron** [comment]
Should I look in any particular log file? I just launched your build, tried accessing recent documents from several applications by using right-arrow, then opened Console.app and looked for 'GET_RECENTS', which yields no results.

<img width="635" alt="temp" src="https://user-images.githubusercontent.com/33901276/37095767-14ed2628-2217-11e8-8d10-105f47bd79f0.png">

**Comment by skurfer** [comment]
You’re looking in the right place. That makes me think it never even gets to the part that loads recent documents. Here’s another test build with logging further up in the code.

https://cl.ly/2p1b101H020X

I made sure it was signed this time so you won’t get the “unknown developer” warning.

**Comment by mdaeron** [comment]
Just tried. Again, nothing shows up when searching "ISSUE2431".

**Comment by skurfer** [comment]
What if you try right-arrowing into a folder? First, does it work as expected? And if so, do you see anything in the console?

**Comment by rosano** [comment]
I haven't been able to test the build yet, but I can confirm that right-arrowing into a folder works as expected in the public release.

**Comment by mdaeron** [comment]
Same here. Right-arrowing into folders always works, right-arrowing into application never does.

No "ISSUE2431" in the console in either case.

**Comment by skurfer** [comment]
If I right-arrow into downloads with that test build, I see

```
default	11:57:27.613787 -0500	Quicksilver	ISSUE2431 load children
default	11:57:27.613888 -0500	Quicksilver	ISSUE2431 single file: /Users/rob/Downloads
default	11:57:27.613929 -0500	Quicksilver	ISSUE2431 this is a directory
```

If it’s working, you should see _something_.

**Comment by mdaeron** [comment]
My bad, I'm indeed getting these messages:

ISSUE2431 single file: /Users/mathieu/Desktop
ISSUE2431 this is a directory
ISSUE2431 load children

What happened earlier was I had two different instances of Quicksilver running, and one of them was the official build.

**Comment by skurfer** [comment]
OK. What do you see when right-arrowing into an application?

**Comment by mdaeron** [comment]
Nothing at all.

**Comment by rosano** [comment]
I tried manually navigating from the Applications folder with Quicksilver and did not see any log when right-arrowing into Preview, but there was a disclosure arrow for Quicksilver and there was a log when right-arrowing:
```
default	15:07:41.438307 -0500	Quicksilver	ISSUE2431 load children
default	15:07:41.438415 -0500	Quicksilver	ISSUE2431 single file: /Applications
default	15:07:41.438479 -0500	Quicksilver	ISSUE2431 this is a directory
default	15:08:11.150098 -0500	Quicksilver	ISSUE2431 load children
default	15:08:11.150203 -0500	Quicksilver	ISSUE2431 single file: /Applications/Quicksilver.app
default	15:08:11.150267 -0500	Quicksilver	ISSUE2431 this is a directory
default	15:08:11.151253 -0500	Quicksilver	ISSUE2431 this is an application
default	15:08:11.151334 -0500	Quicksilver	ISSUE2431 found child preset: QSPresetQSCatalogEntries
```

**Comment by skurfer** [comment]
OK. Here’s another build with more logging.

https://cl.ly/1M3Y2w1h1M1B

If anyone has Xcode and would be willing to troubleshoot that way, this might go faster. (I would provide more detail about where and what to look for.)

**Comment by mdaeron** [comment]
Now when I type 'prev' for Preview, here's what I get:

~~~~
default	15:06:54.989353 +0100	Quicksilver	ISSUE2431 checking Preview.app for children
default	15:06:54.992468 +0100	Quicksilver	ISSUE2431 checking A Compact Underwater Vehicle: The Nanoseeker | Hackaday for children
default	15:06:54.994543 +0100	Quicksilver	ISSUE2431 checking ImageCaptureService.app for children
default	15:06:54.995181 +0100	Quicksilver	ISSUE2431 checking SpeechService.service.app for children
default	15:06:54.997070 +0100	Quicksilver	ISSUE2431 checking SpeechRecognitionServer.app for children
default	15:06:54.997733 +0100	Quicksilver	ISSUE2431 checking Desktop & Screen Saver for children
default	15:06:55.083654 +0100	Quicksilver	ISSUE2431 checking Preview.app for children
default	15:06:59.317249 +0100	Quicksilver	ISSUE2431 right arrow
default	15:06:59.317312 +0100	Quicksilver	ISSUE2431 checking Preview.app for children
default	15:06:59.324765 +0100	Quicksilver	ISSUE2431 checking Preview.app for children
default	15:06:59.327498 +0100	Quicksilver	ISSUE2431 checking A Compact Underwater Vehicle: The Nanoseeker | Hackaday for children
default	15:06:59.330365 +0100	Quicksilver	ISSUE2431 checking ImageCaptureService.app for children
default	15:06:59.331347 +0100	Quicksilver	ISSUE2431 checking SpeechService.service.app for children
default	15:06:59.333835 +0100	Quicksilver	ISSUE2431 checking SpeechRecognitionServer.app for children
default	15:06:59.334523 +0100	Quicksilver	ISSUE2431 checking Desktop & Screen Saver for children
default	15:06:59.386787 +0100	Quicksilver	ISSUE2431 checking Preview.app for children
default	15:06:59.401068 +0100	Quicksilver	ISSUE2431 checking A Compact Underwater Vehicle: The Nanoseeker | Hackaday for children
default	15:06:59.403131 +0100	Quicksilver	ISSUE2431 checking ImageCaptureService.app for children
default	15:06:59.403766 +0100	Quicksilver	ISSUE2431 checking SpeechService.service.app for children
default	15:06:59.405705 +0100	Quicksilver	ISSUE2431 checking SpeechRecognitionServer.app for children
default	15:06:59.406387 +0100	Quicksilver	ISSUE2431 checking Desktop & Screen Saver for children
~~~~

So Quicksilver is detecting my right arrow, but not acting on it, I guess.

**Comment by skurfer** [comment]
I’ve added more logging to a different place, but in the process might have found the problem.

https://cl.ly/282w421u3X04

Let me know if it suddenly works, and if so, which interface you use because I can’t think of any other explanation for why it would only be broken for some people.

**Comment by skurfer** [comment]
New theory: Those of us that still have all the `.sfl` files laying around from older OSes pass the “has recent documents” test, but freshly installed systems will only have `.sfl2` files and QS won’t think it has children.

**Comment by rosano** [comment]
Seems to work for me! Preview, Numbers, Pages - all showing recent documents now.

**Comment by mdaeron** [comment]
Same here. This seems to solve the problem for me.
Thanks to rosano for reviving the issue, and to skurfer for taking the time to work on it.

**Comment by mdaeron** [comment]
Addendum: for some reason (non-standard API probably?) it still doesn't work for BBEdit. Not a deal breaker as far as I'm concerned, but I thought I should mention it.

**Comment by skurfer** [comment]
I’ve released Quicksilver 1.5.9 which should fix this (without all the logging). I’m not sure what’s going on with BBEdit. It’s hard to imagine they’d do something non-standard, but I don’t know what else would explain it.

**Comment by rosano** [comment]
Official release works great - Thank you Rob! 🎉
